component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -296,8 +296,15 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
         if (userToken.equals(contextToken)) {
             context.setProperty(EmailOTPAuthenticatorConstants.OTP_TOKEN, "");
             context.setProperty(EmailOTPAuthenticatorConstants.EMAILOTP_ACCESS_TOKEN, "");
-            String emailFromProfile = context.getProperty(EmailOTPAuthenticatorConstants.RECEIVER_EMAIL).toString();
-            context.setSubject(AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(emailFromProfile));
+            Map<Integer, StepConfig> stepConfigMap = context.getSequenceConfig().getStepMap();
+            AuthenticatedUser authenticatedUser = null;
+            for (StepConfig stepConfig : stepConfigMap.values()) {
+                authenticatedUser = stepConfig.getAuthenticatedUser();
```

Comment:
Since the authenticatedUser is being set in the Authentication context from the identity-extension-utils [1], I hope that we can reuse the method in the identity-extension-utils and retrieve the authenticated user from authentication context instep of iterating through the step configs. Do you see any limitations to retrieved from the authentication context? 

[1] https://github.com/wso2-extensions/identity-extension-utils/blob/master/component/helper/src/main/java/org/wso2/carbon/extension/identity/helper/FederatedAuthenticatorUtil.java#L422

Reply:
Fixed: 2bc937312bd2e41b01ec55aea2cc1d15f4bdc2a0

Reply:
The flow is not going through [1] when the `usecase` property is not set. The documentation for emailOTP does not contain any records on adding any properties. Therefore, this approach is preferred. 
Meantime [2] needs to be updated according to [3].

[1] - https://github.com/wso2-extensions/identity-extension-utils/blob/master/component/helper/src/main/java/org/wso2/carbon/extension/identity/helper/FederatedAuthenticatorUtil.java#L422
[2] documentation - https://docs.wso2.com/display/IS570/Configuring+Email+OTP
[3] - https://github.com/wso2-extensions/identity-outbound-auth-email-otp/blob/master/docs/config.md#usecase-definitions